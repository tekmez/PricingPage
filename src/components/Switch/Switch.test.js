import React from 'react'
import { render, fireEvent } from '@testing-library/react-native'
import CustomSwitch from './Switch'

describe('CustomSwitch', () => {
  it('renders custom switch component', () => {
    const comp = render(<CustomSwitch />).toJSON()
    expect(comp).toMatchSnapshot()
  })

  it('renders with initial state', () => {
    const { getByTestId } = render(<CustomSwitch />)
    const switchElement = getByTestId('custom-switch')
    expect(switchElement.props.value).toBe(true)
  })

  it('toggles switch state when pressed', () => {
    const { getByTestId } = render(<CustomSwitch />)
    const switchElement = getByTestId('custom-switch')
    expect(switchElement.props.value).toBe(true)
    fireEvent(switchElement, 'onValueChange', false)
    expect(switchElement.props.value).toBe(false)
  })
})
