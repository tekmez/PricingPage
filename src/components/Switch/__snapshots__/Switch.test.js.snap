// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomSwitch renders custom switch component 1`] = `
<RCTSwitch
  accessibilityRole="switch"
  onChange={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  onTintColor="#21D840"
  style={
    [
      {
        "height": 31,
        "width": 51,
      },
      {
        "backgroundColor": "#767577",
        "borderRadius": 16,
      },
    ]
  }
  testID="custom-switch"
  thumbTintColor="#fff"
  tintColor="#767577"
  value={true}
/>
`;
